### PR TITLE
Removes unnecessary postStop #30071

### PR DESCRIPTION
**Pull request by nvollmar**
Verified with the given example that this fixed the exception reported in #30071

### CODE REVIEW

**Comment by johanandren** [comment]
OK TO TEST

**Comment by johanandren** [comment]
Would be good with a test with the repeater, I think the test coverage for future source is in https://github.com/akka/akka/blob/031886a7b32530228f34176cd41bba9d344f43bd/akka-stream-tests/src/test/scala/akka/stream/scaladsl/LazySourceSpec.scala

**Review by johanandren — APPROVED** [review]
LGTM, thanks for adding that test case as well!

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala [hunk 1]
```diff
@@ -436,4 +437,18 @@ class SourceSpec extends StreamSpec with DefaultTimeout {
...
+
+    "not cancel substream twice" in assertAllStagesStopped {
+      val result = Source
+        .futureSource(akka.pattern.after(2.seconds)(Future.successful(Source(1 to 2))))
+        .merge(Source(3 to 4))
+        .take(1)
+        .runWith(Sink.ignore)
+
+      Await.ready(result, 4.seconds)
+      result.value shouldBe Some(util.Success(Done))
```
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala:451]:
```
Await.result(result, 4.seconds) shouldBe Done
```

?

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala (hunk 1 above)
Comment (nvollmar) [reply akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala:451]: Was playing around, will revert that to Await.result

Comment (nvollmar) [reply akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceSpec.scala]: Updated the commit, should be ready to merge

**Comment by patriknw** [comment]
Thanks @nvollmar

**Pull request merged**